core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala

```diff
@@ -79,18 +96,43 @@ sealed trait CypherArg {
 /**
   * A literal string to insert into the Cypher string.
   *
-  * @param template the literal value to insert into the [[Statement.template]].
+  * @param template the literal value to insert into the [[CypherStatement.template]].
   */
-sealed abstract class CypherLiteral(override val template: String) extends CypherArg with Proxy {
+sealed abstract class CypherTemplatePart(override val template: String) extends CypherArg with Proxy {
   override def self: Any = template
 }
 
 /**
+  * An identifier to refer to nodes, relationships, or paths in a pattern.
+  *
+  * @see <a href="http://neo4j.com/docs/stable/cypher-identifiers.html">Cypher Identifiers</a>
+  * @param name the name of the identifier
+  */
+final class CypherIdentifier private (name: String) extends CypherTemplatePart(name)
+object CypherIdentifier {
+
+  val Valid = "^[a-zA-Z][a-zA-Z0-9_]*$".r
+
+  def isValid(literal: String): Boolean = {
+    Valid.findFirstMatchIn(literal).isDefined
+  }
+
+  def apply(literal: String): CypherResult[CypherIdentifier] = {
+    if (isValid(literal)) {
+      CypherResultValid(new CypherIdentifier(literal))
+    } else {
+      CypherResultInvalid(CypherIdentifierInvalidFormat(literal))
+    }
+  }
+}
+
+/**
   * A label to add to either a node or relationship.
   *
+  * @see <a href="http://neo4j.com/docs/stable/graphdb-neo4j.html#graphdb-neo4j-labels">Label Documentation</a>
   * @param name the name of the label (without the preceding colon ':')
   */
-final class CypherLabel private (name: String) extends CypherLiteral(s":$name")
+final class CypherLabel private (name: String) extends CypherTemplatePart(s":$name")
 object CypherLabel {
 
   val Valid = "^[a-zA-Z0-9_]+$".r
```

Comment:
I was having trouble with that. Also, I kinda like the explicit nature of this version. Sometimes I forget if `\w` includes `-` (it doesn't)